posts/renv-complications/index.qmd

```diff
@@ -0,0 +1,131 @@
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c logSumExp.c -o logSumExp.o
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c mean2.c -o mean2.o
+In file included from mean2_lowlevel.h:14,
+                 from mean2.c:9:
+mean2_lowlevel_template.h: In function ‘mean2_int’:
+mean2_lowlevel_template.h:59:13: error: ‘DOUBLE_XMAX’ undeclared (first use in this function); did you mean ‘DBL_MAX’?
+   59 |   if (sum > DOUBLE_XMAX) {
+      |             ^~~~~~~~~~~
+      |             DBL_MAX
+mean2_lowlevel_template.h:59:13: note: each undeclared identifier is reported only once for each function it appears in
+In file included from mean2_lowlevel.h:18,
+                 from mean2.c:9:
+mean2_lowlevel_template.h: In function ‘mean2_dbl’:
+mean2_lowlevel_template.h:59:13: error: ‘DOUBLE_XMAX’ undeclared (first use in this function); did you mean ‘DBL_MAX’?
+   59 |   if (sum > DOUBLE_XMAX) {
+      |             ^~~~~~~~~~~
+      |             DBL_MAX
+make: *** [/usr/lib/R/etc/Makeconf:191: mean2.o] Error 1
+ERROR: compilation failed for package ‘matrixStats’
+```
+
+</details>
+
+The explanation for this error can be found in [the `matrixStats` release notes](https://github.com/HenrikBengtsson/matrixStats/blob/799669fc6de6f55a74cab06cc6a97634aa24ab0e/NEWS.md?plain=1#L111-L113), specifically the section for matrixStats 0.63.0:
+
+> * Updated native code to use the C99 constant `DBL_MAX` instead of legacy S
+constant `DOUBLE_XMAX`, which is planned to be unsupported in R (>= 4.2.0).
+
+## Some solutions
+
+### Larger binary store
+
+We discussed how many issues when using renv can arise during the package compilation from source.
+A potential solution would be to avoid this compilation step and always install pre-compiled binaries.
+
+This is not possible while installing from CRAN as CRAN only provides binaries for recent versions of R and for a limited number of platforms.
+
+But Posit for example provides a larger collection of binaries, for different package versions, and different platforms, via their Public [Posit Package Manager (PPM)](https://packagemanager.posit.co/).
+
+Making sure you install from PPM rather than CRAN can be a first simple step to make some of the issues discussed here vanish.
+
+### Extended reproducibility
+
+Another solution could be to add more complex reproducibility solutions that go beyond the scope of renv.
+
+#### renv with rig
+
+The R version is specified in `renv.lock` and to avoid incompatibility of older package versions with newer versions of R, you could run the declared R version.
```

Comment:
Didn't the post mention `renv` manages package versions and the R version? If so, it may leave the reader confused as to what `rig` really adds. Might it help to clarify the difference?

Reply:
> Didn't the post mention renv manages package versions and the R version?

If some part of the post suggested this, it should be updated. The R version number will be recorded in renv but as an R package itself, renv cannot use this info. It can only manage the version of R packages.

Do you have suggestions on how & where to clarify this?

Reply:
I misread I see now. I will leave a comment where I misread for your reference.